[StefanosAv]iP

StefanosAv opened the pull request.

Commit:
Add Gradle support

Commit:
Add Level-1

Commit:
Add Level-2

Commit:
Add Level-3

Commit:
Add Level-4

Commit:
Add: A-TextUiTesting

Commit:
Add: Level-5

Commit:
Add: Level-6

Yufannnn commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,14 @@
+public class Deadline extends Task {
```
I like how the indentation and placement of braces in this class follow good programming practices, making the code easier to read and understand.  The code is well-structured and easy to read. The class name, variables and methods are named clearly and appropriately. :wink:

Yufannnn commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,144 @@
+import java.util.ArrayList;
+import java.lang.Math;
+
 public class Duke {
```
I like how you have used comments in the code to explain the logic and make it easier to understand. However, any reason why you structured the method this way instead of breaking it down into simpler methods or keeping the methods shorter and more readable? :worried:

Yufannnn commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,144 @@
+                        if (taskList.get(index).isDone()) {
+                            throw new DukeException("marked");
+                        }
+                        else {
```
should the line ```` else { ```` be placed on the same line with the preceding `}`, according to the code standard for if-else statement? :frowning:

Yufannnn commented on src/main/java/DukeException.java:
```diff
@@ -0,0 +1,44 @@
+public class DukeException extends Exception {
```
Any reason why you did it this way instead of using polymorphism or having separate classes for each exception?

Yufannnn commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,36 @@
+public class Task {
```
Any reason why you did not make the Task class abstract since you did not directly instantiate a Task object anyway?

Yufannnn reviewed (commented):
Overall, the code is mostly easy to follow, but there are a few places with overly nested logic. Additionally, there are a few minor deviations from common coding standards. It is great to see that the code is well-commented on, making it easier to understand. Some of your methods are getting lengthy, making them difficult to read and maintain. Consider breaking them down into smaller, simpler methods.

unfazing commented on src/main/java/DukeException.java:
```diff
@@ -0,0 +1,44 @@
+    }
+
+    public void EmptyDescriptionException() {
+        switch (this.task) {
```
You can consider amending the indentation rules for the `case` blocks according to the Java conventions (no indent in front of `case`)

unfazing commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,144 @@
+                        System.out.println(description + " - " + by);
+
+                        if (description.equals("")) {
+                            throw new DukeException("deadline");
```
You may want to include a more specific error message when you throw Exceptions.

unfazing reviewed (commented):
The code is generally readable and follows the coding standard!

Commit:
Add Level-7. Save

Commit:
Add Level-8: Date & Time

Commit:
Add A-MoreOOP

Commit:
Add A-Packages

Commit:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S2/ip

* 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S2/ip:
  Add Gradle support

Commit:
Add A-JUnit tests

Commit:
Add A-JavaDoc

Commit:
Add Level-9: Find

Commit:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Add Level-9: Find

Commit:
Add A-CodingStandard and refactor code for better OOP

Commit:
Add Level-10: GUI

Commit:
Add Assertions

Add assertions to certain methods to define assumptions about the
program state so that the runtime can verify them.

This enables easier debugging and ensures that the code executes as
intended.

Commit:
Add A-CodeQuality

Improve overall code quality and make minor adjustments to GUI

StefanosAv committed:
Merge pull request #2 from StefanosAv/branch-A-Assertions

Add Assertions

StefanosAv committed:
Merge branch 'master' into branch-A-CodeQuality

StefanosAv committed:
Merge pull request #3 from StefanosAv/branch-A-CodeQuality

Add A-CodeQuality

Commit:
Add BCD-Extension: C-Tagging

Commit:
Add BCD-Extension: C-Tagging

StefanosAv committed:
Merge branch 'master' into branch-BCD-Extension

StefanosAv committed:
Merge pull request #4 from StefanosAv/branch-BCD-Extension

Add BCD-Extension: C-Tagging

Commit:
Add minor improvements and error handling

Commit:
Merge branch 'master' of https://github.com/StefanosAv/ip

* 'master' of https://github.com/StefanosAv/ip:
  Add BCD-Extension: C-Tagging

Commit:
Add Ui.png and UG

StefanosAv committed:
Update README.md

Commit:
Merge branch 'master' of https://github.com/StefanosAv/ip

* 'master' of https://github.com/StefanosAv/ip:
  Update README.md

StefanosAv committed:
Update README.md

StefanosAv committed:
Update README.md
